cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -1530,6 +1530,22 @@ private predicate isUnsupportedGuardPhi(Variable v, RangeSsaDefinition phi, Vari
   )
 }
 
+/**
+ * Gets the upper bound of the expression, if the expression is guarded.
+ * An upper bound can only be found, if a guard phi node can be found, and the
+ * expression has only one immediate predecessor.
+ */
+private float getGuardedUpperBound(VariableAccess guardedAccess) {
+  exists(
+    RangeSsaDefinition def, StackVariable v, VariableAccess guardVa, Expr guard, boolean branch
+  |
+    def.isGuardPhi(v, guardVa, guard, branch) and
+    exists(unique(BasicBlock b | b = def.(BasicBlock).getAPredecessor())) and
```

Comment:
Please add a comment explaining why we added this line.

Reply:
Will do.